pkg/kubectl/delete.go

```diff
@@ -203,8 +203,8 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout
 			return err
 		}
 	}
-	falseVar := false
-	deleteOptions := &metav1.DeleteOptions{OrphanDependents: &falseVar}
+	policy := metav1.DeletePropagationForeground
+	deleteOptions := &metav1.DeleteOptions{PropagationPolicy: &policy}
```

Comment:
The reasoning is the same as for generic deletion. Is there a specific reason for deleting ReplicationController immediately?

Reply:
At this point, the reaper has called `scaler.Scale` to stop all the dependent pods, so you can set the policy to "background" to delete the owner immediately. Moving away from `OrphanDependents` is good, as the field is marked deprecated.

Or, you can remove the reapers, and rely on the garbage collector (GC) to do the foreground deletion. The behavior will be different in that:
1. the GC deletion waits for all the dependent objects to be deleted before deleting the replicaset, not limiting to pods;
2. the GC deletion waits for the dependents to be deleted from etcd; while the scalers only wait for the pods to be in the NotReady state. GC's behavior is better, because for objects like pods, only the deletion from etcd indicates the occupied resources are released.
3.  GC cascadingly deletes dependents' dependents; reapers don't.

Moving away from reapers requires more discussion in the sig-cli, and can be done in another PR.

Reply:
Thanks for the context, will change it to "background" there.